### PR TITLE
RibbonClient use replaceFirst to remove host rather than other parts of ...

### PR DESCRIPTION
**Pull request by spencergibb**
...the uri.

 fixes gh-221

### CODE REVIEW

File: core/src/main/java/feign/Request.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
   private final byte[] body;
   private final Charset charset;
 
-  Request(String method, String url, Map<String, Collection<String>> headers, byte[] body,
+  public Request(String method, String url, Map<String, Collection<String>> headers, byte[] body,
```
Comment (spencergibb) [inline core/src/main/java/feign/Request.java]: Quick and dirty so I could create a test that worked, see below

File: ribbon/src/main/java/feign/ribbon/RibbonClient.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public Response execute(Request request, Request.Options options) throws IOExcep
     try {
       URI asUri = URI.create(request.url());
       String clientName = asUri.getHost();
-      URI uriWithoutHost = URI.create(request.url().replace(asUri.getHost(), ""));
+      URI uriWithoutHost = URI.create(request.url().replaceFirst(asUri.getHost(), ""));
```
Comment (spencergibb) [inline ribbon/src/main/java/feign/ribbon/RibbonClient.java]: The actual fix

File: ribbon/src/test/java/feign/ribbon/RibbonClientTest.java [hunk 3]
```diff
@@ -172,6 +175,38 @@ public void testFeignOptionsClientConfig() {
...
+    RibbonClient client = RibbonClient.builder()
+        .delegate(new Client() {
+          @Override
+          public Response execute(Request request, Request.Options options) throws IOException {
+            holder.held = request.url();
+            return Response.create(200, "", Collections.EMPTY_MAP, "success", Charset.defaultCharset());
+          }
+        }).build();
+
+    client.execute(new Request("GET", "http://"+client()+"/"+client(), Collections.EMPTY_MAP, new byte[0], Charset.defaultCharset()), new Request.Options(1111, 22222));
```
Comment (spencergibb) [inline ribbon/src/test/java/feign/ribbon/RibbonClientTest.java]: Suggestions on a better way to test this?  One thought was to bring in power mock to mock the final `Request`.  The other would be to refactor the replaceFirst line into a static method that was easy to test.

**Comment by spencergibb** [comment]
@adriancole any chance you could review?

**Comment by adriancole** [comment]
Yep. Was on my plan after I finish coffee :) I typically do
feign/denominator stuff between 7:15-8:15 pacific time daily.

**Comment by spencergibb** [comment]
@adriancole thanks.  Sometimes I forget what timezone *I'm* in.

**Comment by adriancole** [comment]
squash into one commit? otherwise LGTM

**Comment by adriancole** [comment]
ps what timezone are you in? :)

**Comment by spencergibb** [comment]
I'm in the East until July, then back to my normal Mountain.  My git foo is failing me this morning, pointers?

**Comment by spencergibb** [comment]
alright, looks like a squashed everything.

**Comment by adriancole** [comment]
thanks for the fix!

**Comment by spencergibb** [comment]
thanks! any ideas when you'll release 8.2?

**Comment by adriancole** [comment]
Isn't hard. It seems more like a subminor if only thing is this fix. Could
do this weekend.

**Comment by spencergibb** [comment]
sweet.  I'm great if it's 8.1.1 and not 8.2.

**Pull request merged**